screenapiexample/src/main/java/net/gini/android/vision/screen/CameraScreenApiActivity.java

```diff
@@ -1,27 +1,159 @@
+            }
+            if (hasMoreThan5MB(uri)) {
+                callback.documentRejected(getString(R.string.document_size_too_large));
+                return;
+            }
+            // IMPORTANT: always call one of the callback methods
+            if (isJpegOrPdf(uri)) {
                 callback.documentAccepted();
             } else {
-                callback.documentRejected("Diese Datei ist leider größer als 5MB.");
+                callback.documentRejected(getString(R.string.unsupported_document_type));
             }
         } else {
-            callback.documentRejected(getString(R.string.gv_document_import_error));
+            // IMPORTANT: always call one of the callback methods
+            callback.documentAccepted();
+        }
+    }
+
+    private boolean hasMoreThan5MB(final Uri uri) {
+        final int fileSize = UriHelper.getFileSizeFromUri(uri, this);
+        return fileSize > 5 * 1024 * 1024;
+    }
+
+    private boolean isJpegOrPdf(final Uri uri) {
+        InputStream inputStream = null;
+        try {
+            inputStream = getContentResolver().openInputStream(uri);
+            if (inputStream != null) {
+                final byte[] magicBytes = new byte[4];
+                final int read = inputStream.read(magicBytes);
+                inputStream.reset();
+                return read != -1
+                        && (isJpegWithExif(inputStream, magicBytes, read)
+                        || isPDF(uri, magicBytes, read));
+            }
+        } catch (FileNotFoundException e) {
+            LOG.error("Could not open document", e);
+            return false;
+        } catch (IOException e) {
+            LOG.error("Could not read document", e);
+            return false;
+        } finally {
+            if (inputStream != null) {
+                try {
+                    inputStream.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean isJpegWithExif(final InputStream inputStream, final byte[] magicBytes,
+            final int read)
+            throws IOException {
+        return isDecodableToBitmap(inputStream)
+                && read == 4
+                && magicBytes[0] == (byte) 0xFF
+                && magicBytes[1] == (byte) 0xD8
+                && magicBytes[2] == (byte) 0xFF
+                && magicBytes[3] == (byte) 0xE1;
+    }
+
+    private boolean isDecodableToBitmap(final InputStream inputStream)
+            throws IOException {
+        BitmapFactory.Options options = new BitmapFactory.Options();
+        final Bitmap bitmap = BitmapFactory.decodeStream(inputStream, new Rect(), options);
+        return bitmap != null;
+    }
+
+    private boolean isPDF(final Uri uri, final byte[] magicBytes, final int read) {
+        boolean isRenderablePdf = true;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```

Comment:
Since Android KitKat or below does not have PDFRenderer class, how do you know that the PDF is always renderable? Or is it only for the custom check?

Reply:
If I would return false by default then no PDF would be accepted on KitKat or below so I do the optimist. And it's just an example...